static/app/components/u2f/u2finterface.tsx

```diff
@@ -116,11 +120,19 @@ class U2fInterface extends Component<Props, State> {
           () => {
             const u2fResponse = this.getU2FResponse(data);
             const challenge = JSON.stringify(this.props.challengeData);
+            const authStateData = this.props.authStateData
+              ? JSON.stringify(this.props.authStateData)
+              : this.props.authStateData;
 
             if (this.state.responseElement) {
               // eslint-disable-next-line react/no-direct-mutation-state
               this.state.responseElement.value = u2fResponse;
             }
+            if (this.state.authStateElement && authStateData) {
+              this.setState({
+                authStateElement: authStateData as unknown as HTMLInputElement,
```

Comment:
oh it doesn't let me cast it directly to `HTMLInputElement` b/c there's some error saying there's no overlap btwn the classes. I poked around and it looks like we do this in some other parts of the code.
`authStateData` is a JSON that's been stringified

Reply:
Oof... 😬 🤔 

well... it _does_ seem like this _shouldn't_ be an `HtmlInputElement` right?
`authStateData` is a jsonified string? is this being converted into an element somewhere??

rather than perpetuating this hack - I would prefer we `@ts-ignore` this particular error until we properly set the value?

Reply:
I'm not sure how this `HtmlInputElement` stuff works, but I think we need it so that our view which calls a django html template which renders this component can send that data back in the response body

Reply:
I can `@ts-ignore` it for now

Reply:
Ohhh uggh...

So this is an element being injected into a template
who's results are now being written back to state? 🤔

Reply:
Responded in slack

but it looks like here we were attempting to copy the previous ref element example
```
this.state.responseElement.value = u2fResponse;
```

but rather than setting the element value, we're attempting to replace the entire element.

This looks incorrect?